### PR TITLE
Student registration page- data from backend using "/api/groups"

### LINKED ISSUE #215: Frontend: student registration page
Instead of hard-coding the groups, use `/api/groups` to get the list of groups for the dropdown. While the API is loading, show `Loader`.

Once the user clicks Submit, use the API in #103 to perform registration and navigate to `/` after the call ends.

### PR DESCRIPTION
**Pull request by AnilMula**
# Description

This PR is to get the data from the backend using endpoint /api/groups into the student component. I had to change few names and prop types.

Fixes #215 

# How to test?
It is tested with storybook by providing "http://localhost:3000/api/groups " in axios.get(), insted of "/api/groups". 

# Checklist

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] This PR is ready to be merged and not breaking any other functionality

### CODE REVIEW

File: src/client/containers/RegistrationPage/StudentRegistrationPage/StudentRegistrationPage.component.js [hunk 1]
```diff
@@ -1,33 +1,33 @@
...
 export const StudentRegistrationPage = () => {
   const [name, setName] = useState('');
   const [hyfclass, sethyfClass] = useState('Class14');
+  const [groups, setGroups] = useState();
+  const [isLoading, setIsLoading] = useState(true);
 
+  useEffect(() => {
+    axios.get('/api/groups').then((response) => {
+      setGroups(response.data);
+      setIsLoading(false);
```
Comment (tpetrina) [inline src/client/containers/RegistrationPage/StudentRegistrationPage/StudentRegistrationPage.component.js:25]: Also add `sethyfClass(response.data[0].id)`.

File: src/client/components/Dropdown/DropDown.js [hunk 2]
```diff
@@ -24,7 +24,7 @@ DropDown.propTypes = {
   setValue: PropTypes.func,
   items: PropTypes.arrayOf(
     shape({
-      id: PropTypes.string,
+      id: PropTypes.number,
```
Comment (tpetrina) [inline src/client/components/Dropdown/DropDown.js:27]: Are you sure this change didn't break existing usages of `Dropdown` like in storybook or elsewhere?

**Review by tpetrina — CHANGES_REQUESTED** [review]
Pretty good, just check if changes didn't break anything else.

File: src/client/containers/RegistrationPage/StudentRegistrationPage/StudentRegistrationPage.component.js (hunk 1 above)
Comment (AnilMula) [reply src/client/containers/RegistrationPage/StudentRegistrationPage/StudentRegistrationPage.component.js:25]: updated

File: src/client/components/Dropdown/DropDown.js (hunk 2 above)
Comment (AnilMula) [reply src/client/components/Dropdown/DropDown.js:27]: I had a look over all the components in storybook,  it doesn't seen anything like breaking. In database id is number for all the tables, so this is the reason I have changed PropType to number

**Review by tpetrina — APPROVED** [review]
Good stuff!

**Review by Deepti-Dk — APPROVED** [review]

**Pull request merged**